omero/sysadmins/unix/server-debian10-ice36.rst

```diff
@@ -79,7 +79,7 @@ Installing OMERO.server
 
 **The following step is run as root.**
 
-We recommend to create a virtual environment and install the Ice Python binding and PyTables_ using ``pip``:
+We recommend to create a virtual environment and install the Ice Python binding, PyTables_ and ``omero-certificates`` using ``pip``:
```

Comment:
I wonder if it would be worth putting an `omero-server` package in place which depends on all the recommended items.

Reply:
We will need one per linux flavour
Something to look into after. I will create an issue

Reply:
Why? Because of the PyTables version? I think that can be special cased in the setup.py itself.

Reply:
https://www.python.org/dev/peps/pep-0508/#environment-markers

Reply:
no the python binding

Reply:
My idea would be to create a new package on pypi `omero-server` which initially may not even have any code in it though it could have the admin plugins for example. Importantly, though, it would depend on:
 - pytables
 - omero-py
 - omero-certificates
 - omero-metadata
 - ...whatever other plugins we think should be installed

That way we wouldn't need to update the documentation list each time but only push out a new package. We _would_ then need to add to the upgrade documentation `pip install -U omero-server` but that would also pull in new omero-py versions.

Reply:
You could make it have exact versions, add dependabot to automatically bump them, then this is a potential solution for https://github.com/ome/design/issues/103. Want to add it there?

Reply:
Added, but I'd say let's do that _eventually_ once we know more about what all it entails, and instead focus here on just simplifying the users' experience on installation. e.g. if we would encode a minimum or maximum version in the docs due to an incompatibility (which we've needed to do a few times) then we would capture that in `omero-server`'s `setup.py`, but not go further until 103 is worked out.

Reply:
The question for now is: do we want to use ``omero-certificates`` in our doc instead of the current patch ``omero config set omero.glacier2.IceSSL.Ciphers HIGH:ADH:@SECLEVEL=0``. If not I will close the 2 PRs (doc and omero-install)

Reply:
On that question, I'm for it. But I'm not for needing users to know more and more CLI plugins that need to be manually installed. E.g. this PR has the same package copied 4 times and if we add it and then remove it, it may lead to confusion.

Reply:
I could make it so it is only one file in omero-install (I used an existing file for convenience) from where the snippet is coming from. The patch for example is one file.

It will always appear  X times in the doc due to the nature of the guides.

The ssl issue/question is coming at regular intervals.
Any new version we support will have the issues e.g. ubuntu 20.04/Centos 8.
 I don't disagree with you but we need to be ready to repeat ourselves on the ssl question and recommend to install that specific plugin over and over.

There is no magic solution until we have everything in place. The same is true for other plugins, I can imagine a "visualisation" install for web, same at cli levels (the new roi is a tool that is needed). We do not have that in place but already recommend some of those apps in our doc.

Reply:
See https://github.com/ome/design/issues/105

Reply:
name of the meta-package should probably be something more explicit like ``omero-server-deps``

Reply:
The only 2 platforms to cap ``tables`` will be debian9 and ubuntu16.04. 
Both are still in the doc with guides unless we decide to "drop" them
Only debian9 needs the "ssl" patch. so we could leave 16.04 doc as it is we do not want to add too many checks in setup.

Reply:
debian10
```
 /opt/omero/server/venv3/bin/pip install git+https://github.com/joshmoore/omero-cli-server.git@master
Collecting git+https://github.com/joshmoore/omero-cli-server.git@master
  Cloning https://github.com/joshmoore/omero-cli-server.git (to revision master) to /tmp/pip-req-build-9ry2r_3j
Collecting omero-certificates (from omero-server==5.6.0.dev0)
  Downloading https://files.pythonhosted.org/packages/2f/76/9467089a123fb2d8fb8a0c8b0e08ca5201082a77455261c23ea511b85ab5/omero_certificates-0.2.0-py3-none-any.whl
Collecting omero-py (from omero-server==5.6.0.dev0)
  Downloading https://files.pythonhosted.org/packages/4d/9b/6e56e39a5df35489a105121da1bd60aba10778e79fc8a29d987659579242/omero_py-5.6.2-py3-none-any.whl (2.8MB)
Collecting tables (from omero-server==5.6.0.dev0)
  Downloading https://files.pythonhosted.org/packages/0f/cb/4097be890a773af95343389faa8c283b0d9ff606f144227a548461dcbdd5/tables-3.6.1-cp37-cp37m-manylinux1_x86_64.whl (4.3MB)
Collecting Pillow (from omero-py->omero-server==5.6.0.dev0)
  Downloading https://files.pythonhosted.org/packages/ab/f8/d3627cc230270a6a4eedee32974fbc8cb26c5fdb8710dd5ea70133640022/Pillow-7.1.2-cp37-cp37m-manylinux1_x86_64.whl (2.1MB)
Collecting numpy (from omero-py->omero-server==5.6.0.dev0)
  Downloading https://files.pythonhosted.org/packages/1f/df/7988fbbdc8c9b8efb575029498ad84b77e023a3e4623e85068823a102b1d/numpy-1.18.4-cp37-cp37m-manylinux1_x86_64.whl (20.2MB)
Collecting future (from omero-py->omero-server==5.6.0.dev0)
  Downloading https://files.pythonhosted.org/packages/45/0b/38b06fd9b92dc2b68d58b75f900e97884c45bedd2ff83203d933cf5851c9/future-0.18.2.tar.gz (829kB)
Requirement already satisfied: zeroc-ice<3.7,>=3.6.4 in /opt/omero/server/venv3/lib/python3.7/site-packages (from omero-py->omero-server==5.6.0.dev0) (3.6.5)
Collecting numexpr>=2.6.2 (from tables->omero-server==5.6.0.dev0)
  Downloading https://files.pythonhosted.org/packages/c5/de/3811c7ce8612b4a27bfbf81a779c5970a31a038e32dc6a979e7f2867f56d/numexpr-2.7.1-cp37-cp37m-manylinux1_x86_64.whl (162kB)
Building wheels for collected packages: omero-server, future
  Running setup.py bdist_wheel for omero-server: started
  Running setup.py bdist_wheel for omero-server: finished with status 'error'
  Complete output from command /opt/omero/server/venv3/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-req-build-9ry2r_3j/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/pip-wheel-3_sv88ty --python-tag cp37:
  usage: -c [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: -c --help [cmd1 cmd2 ...]
     or: -c --help-commands
     or: -c cmd --help
  
  error: invalid command 'bdist_wheel'
  
  ----------------------------------------
  Failed building wheel for omero-server
  Running setup.py clean for omero-server
  Running setup.py bdist_wheel for future: started
  Running setup.py bdist_wheel for future: finished with status 'error'
  Complete output from command /opt/omero/server/venv3/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-v4j0le_2/future/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/pip-wheel-pnlhzyu6 --python-tag cp37:
  usage: -c [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: -c --help [cmd1 cmd2 ...]
     or: -c --help-commands
     or: -c cmd --help
  
  error: invalid command 'bdist_wheel'
  
  ----------------------------------------
  Failed building wheel for future
  Running setup.py clean for future
Failed to build omero-server future
Installing collected packages: Pillow, numpy, future, omero-py, omero-certificates, numexpr, tables, omero-server
  Running setup.py install for future: started
    Running setup.py install for future: finished with status 'done'
  Running setup.py install for omero-server: started
    Running setup.py install for omero-server: finished with status 'done'
Successfully installed Pillow-7.1.2 future-0.18.2 numexpr-2.7.1 numpy-1.18.4 omero-certificates-0.2.0 omero-py-5.6.2 omero-server-5.6.0.dev0 tables-3.6.1

```

Reply:
Reading this entire thread as well as the issue a few comments & questions:

- in general :+1: for https://github.com/ome/design/issues/105 as well as promoting `omero-certificates` in our sysadmin doc given the recurring SSL issues our community is experiencing in various platforms
- as we are adding more components, :+1: for a meta-package that aims to simplify our maintenance of the documentation and streamline the installation process for sysadmins
- re name, I am not overly convinced by `omero-server-deps`. If I understand this meta-package would pull `omero-py`, `omero-certificates` maybe the DB and admin plugins too, it will effectively be the full Python component of OMERO.server while `omero-server-deps` reads more like pulling external dependencies (Ice, PyTables)
- one question related to the overall target and deadline of this work. Whether we add a meta-package or just list `omero-certificates`, this is a significative change to our sysadmin instructions as we are introducing a new dependency, even if only recommended for now. How would the amended documentation be released?

Reply:
It pulls both "internal" and external e.g. tables

Reply:
In terms of deadline, the suggested "patch" for open-ssl does not always work.
So updating the doc to use omero-certificates makes more sense for the next doc release. (sooner the better since c7 doc is broken due to pg url)
Then we start to have more dependencies to the list started the discussion of the meta-package.  Adding a new entry to the meta-package e.g. ``scripts``, will not require a release of omero-install nor the doc